### PR TITLE
Adds ignore-running flag to pr,tr delete cmd

### PR DESCRIPTION
**Pull request by pradeepitm12**
This is copy of #1436
Adds ignore-running to pipelinerun and taskrun delete command
this ignores all the running and pending resources while delete

Signed-off-by: Pradeep Kumar <pradkuma@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Run the code checkers with `make check`
- [ ] Regenerate the manpages, docs and go formatting with `make generated`
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

### CODE REVIEW

**Comment by pradeepitm12** [comment]
cc: @vinamra28

File: pkg/cmd/pipelinerun/delete_test.go [hunk 1]
```diff
@@ -239,7 +239,7 @@ func TestPipelineRunDelete(t *testing.T) {
 			input:       seeds[0].pipelineClient,
 			inputStream: strings.NewReader("y"),
 			wantError:   false,
-			want:        "Are you sure you want to delete all PipelineRuns related to Pipeline \"pipeline\" (y/n): All PipelineRuns associated with Pipeline \"pipeline\" deleted in namespace \"ns\"\n",
+			want:        "Are you sure you want to delete all PipelineRuns related to Pipeline \"pipeline\" (y/n): All PipelineRuns associated with Pipeline \"pipeline\" deleted in namespace \"ns\" ignores running or pending\n",
```
Comment (pradeepitm12) [inline pkg/cmd/pipelinerun/delete_test.go:242]: should we really consider adding `ignores running or pending`?

**Comment by pradeepitm12** [comment]
/hold
while testing below fails 
```
./tkn pr delete  --ignore-running false --all -f
Error: --all flag should not have any arguments or flags specified with it
```
and 
```
./tkn pr delete  --ignore-running=false --all -f
All PipelineRuns deleted in namespace "default"
```

let's get this merge https://github.com/tektoncd/cli/pull/1443, as it takes care of some of the flags and its combination
cc: @piyush-garg @vinamra28

**Comment by piyush-garg** [comment]
Do we need to mention `ignores running or pending` in output? I don't think so that we need to explicitly mention that. That can be flag description but not in output after success.

**Comment by pradeepitm12** [comment]
> Do we need to mention `ignores running or pending` in output? I don't think so that we need to explicitly mention that. That can be flag description but not in output after success.

I should have written this with more description. I too feel the same. 
> should we really consider adding ignores running or pending?
The only thing is to ignore the running tr/pr is by default enabled. So even if the user does not specify the flag running ones will be ignored.

**Comment by pradeepitm12** [comment]
I would really be happy to remove those extra details.

File: pkg/cmd/pipelinerun/delete_test.go [hunk 2]
```diff
@@ -266,7 +266,7 @@ func TestPipelineRunDelete(t *testing.T) {
 			input:       seeds[4].pipelineClient,
 			inputStream: nil,
 			wantError:   false,
-			want:        "All but 2 PipelineRuns deleted in namespace \"ns\"\n",
+			want:        "All but 2 PipelineRuns(Completed) deleted in namespace \"ns\"\n",
```
Comment (pradeepitm12) [inline pkg/cmd/pipelinerun/delete_test.go:269]: @piyush-garg @vinamra28 
how about this  `(Completed)`

Comment (vinamra28) [reply pkg/cmd/pipelinerun/delete_test.go:269]: @pradeepitm12 there can be a case where the pipeline is in error state so should we show the same for that as well?

**Review by piyush-garg — APPROVED** [review]

File: pkg/cmd/pipelinerun/delete_test.go (hunk 2 above)
Comment (piyush-garg) [reply pkg/cmd/pipelinerun/delete_test.go:269]: Completed mean both to me - either success or failure, but it has completed running

Comment (piyush-garg) [reply pkg/cmd/pipelinerun/delete_test.go:269]: cc @vdemeester @chmouel Any thoughts for this?

**Comment by vinamra28** [comment]
LGTM

File: pkg/cmd/pipelinerun/delete_test.go (hunk 2 above)
Comment (chmouel) [reply pkg/cmd/pipelinerun/delete_test.go:269]: Maybe "Finished" but it's nitpick...

either I am not so fond of having another switch and instead would rather have a filter capability...

i.e:

```
--ignore-filter running,label=key,anotherlabel=another=key,failed,completed
```

which would ignore all of this

we would need to include a `--list-filter` too.

Comment (piyush-garg) [reply pkg/cmd/pipelinerun/delete_test.go:269]: As discussed in WG yesterday, is it fine to with current implementation and refactor later based on other use cases cc @vdemeester @chmouel

**Pull request merged**